src/vector/geometry.rs

```diff
@@ -339,6 +339,92 @@ impl Geometry {
             gdal_sys::OGR_G_AssignSpatialReference(self.c_geometry(), spatial_ref.to_c_hsrs())
         };
     }
+
+    /// Computes and returns the bounding envelope for this geometry
+    pub fn envelope(&self) -> gdal_sys::OGREnvelope {
+        let mut envelope = OGREnvelope {
+            MinX: 0.0,
+            MaxX: 0.0,
+            MinY: 0.0,
+            MaxY: 0.0,
+        };
+        unsafe { gdal_sys::OGR_G_GetEnvelope(self.c_geometry(), &mut envelope) };
+        envelope
+    }
+
+    /// Computes and returns the bounding envelope (3D) for this geometry
+    pub fn envelope3d(&self) -> gdal_sys::OGREnvelope3D {
+        let mut envelope = OGREnvelope3D {
+            MinX: 0.0,
+            MaxX: 0.0,
+            MinY: 0.0,
+            MaxY: 0.0,
+            MinZ: 0.0,
+            MaxZ: 0.0,
+        };
+        unsafe { gdal_sys::OGR_G_GetEnvelope3D(self.c_geometry(), &mut envelope) };
+        envelope
+    }
+
+    /// Determines whether two geometries intersect.
+    /// If GEOS is enabled, then this is done in rigorous fashion otherwise TRUE is returned if the envelopes (bounding boxes)
+    /// of the two geometries overlap.
+    pub fn intersects(&self, other: &Geometry) -> bool {
+        unsafe { gdal_sys::OGR_G_Intersects(self.c_geometry(), other.c_geometry()) == 1 }
+    }
+
+    /// Tests if this geometry and the other geometry are disjoint.
+    /// Geometry validity is not checked. In case you are unsure of the validity of the input geometries,
+    /// call [`is_valid()`] before, otherwise the result might be wrong.
+    #[cfg(have_geos)]
```

Comment:
Outside of a sys-crate, I'm surprised for methods to become defined or not based on something other than a feature flag.

What do people think of something like:

1. Add a `geos` feature to the gdal crate.
2. Make these methods contingent on the `geos` feature.
3. If the `geos` feature is enabled, assert `cfg(have_geos)` is true.

I know the end result is the same, but I think users will have a better chance of guessing how things are supposed to work this way.